src/main/kotlin/com/exactpro/th2/lwdataprovider/handlers/QueueMessagesHandler.kt

```diff
@@ -57,8 +59,18 @@ class QueueMessagesHandler(
         executor.execute {
             val param = CradleGroupRequest(false)
             val streamInfo = ProviderStreamInfo()
-            createSink(handler, streamInfo, batchMaxSize) { bath, attribute ->
-                router.send(bath, attribute, request.externalQueue, QueueAttribute.RAW.value)
+            val batchMetadata = MessageGroupBatchMetadata.newBuilder().setExternalQueue(request.externalQueue).build()
+            createSink(handler, streamInfo, batchMaxSize) { batch, attribute ->
+                val toSend: MessageGroupBatch = batch.setMetadata(batchMetadata)
+                    .build()
+                if (request.sendRawDirectly) {
+                    router.sendExclusive(request.externalQueue, toSend)
+                }
+                if (useAttributes) {
+                    router.send(toSend, attribute, QueueAttribute.RAW.value)
```

Comment:
Maybe use sendAll too

Reply:
Shouldn't we have a single pin matched the attributes in this case?